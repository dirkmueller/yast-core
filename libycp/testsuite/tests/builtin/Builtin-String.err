Parsed:
----------------------------------------------------------------------
"** size **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
size ("")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
size ("linux")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** plus **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
("YaST" + "2")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
("YaST" + 2)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
("YaST" + .two)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
("YaST" + `two)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/builtin/Builtin-String.ycp"
    textdomain "base";
    return (_("YaST") + "2");
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/builtin/Builtin-String.ycp"
    textdomain "base";
    return ("The " + _("YaST"));
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** equality **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
("YaST2" == "YaST2")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
("YaST2" == "yast2")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** inequality **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
("YaST2" != "yast2")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
("YaST2" != "YaST2")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** issubstring **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
issubstring ("some text", "tex")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
issubstring ("some text", "no")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** substring **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
substring ("some text", 5)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
substring ("some text", 42)
----------------------------------------------------------------------
[Interpreter] tests/builtin/Builtin-String.ycp:48 Substring index out of range
Parsed:
----------------------------------------------------------------------
substring ("some text", 5, 2)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
substring ("some text", 42, 2)
----------------------------------------------------------------------
[Interpreter] tests/builtin/Builtin-String.ycp:51 Substring index out of range
Parsed:
----------------------------------------------------------------------
"** mergestring **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
mergestring (["", "abc", "dev", "ghi"], "/")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
mergestring (["abc", "dev", "ghi", ""], "/")
----------------------------------------------------------------------
[Interpreter] tests/builtin/Builtin-String.ycp:58 No match for 'mergestring : <unspec> (list, string)'
[Interpreter] tests/builtin/Builtin-String.ycp:58 Please fix parameter types to match one of:
[Interpreter] tests/builtin/Builtin-String.ycp:58 'mergestring : string (const list <string>, string)' 
[Parser] tests/builtin/Builtin-String.ycp:58 Wrong parameters in call to mergestring(...)
[Parser] tests/builtin/Builtin-String.ycp:58 Expected '(const list <string>, string)', seen '(list, string)'.
Parsed:
----------------------------------------------------------------------
Parser error
----------------------------------------------------------------------
[Interpreter] tests/builtin/Builtin-String.ycp:58 Parser error
Parsed:
----------------------------------------------------------------------
mergestring ([], ".")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
mergestring (["abc", "dev", "ghi"], "")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
mergestring (["abc", "dev", "ghi"], "123")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** deletechars **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
deletechars ("1a2b3c", "abc")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
deletechars ("1a2b3c", "")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
deletechars ("", "abc")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
deletechars ("", "")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** filterchars **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
filterchars ("1a2b3c", "abc")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
filterchars ("1a2b3c", "")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
filterchars ("", "abc")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
filterchars ("", "")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** crypt  -  disabled **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** tolower and toascii **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
tolower ("aBcDeF")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
toascii ("a\326B\304c")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** find et.al. **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
find ("abcdefghi", "efg")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
find ("aaaaa", "z")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findfirstnotof ("abcdefghi", "abcefghi")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findfirstnotof ("aaaaa", "a")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findfirstof ("abcdefghi", "cxdv")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findfirstof ("aaaaa", "z")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findlastof ("abcdecfghi", "cxdv")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findlastof ("aaaaa", "z")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** regex **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpmatch ("aaabbb", "ab") == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpmatch ("aaabbb", "ba") == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpmatch ("aaabbb", ".*ab.*") == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpmatch ("aaabbb", ".*ba.*") == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("aaabbb", "(.*ab).*", "s_\\1_e") == "s_aaab_e")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("aaabbb", "(.*abb)(.*)", "s_\\1_e\\2") == "s_aaabb_eb")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexptokenize ("aaabbBb", "(.*[A-Z]).*") == ["aaabbB"])
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexptokenize ("aaabbb", "(.*ab)(.*)") == ["aaab", "bb"])
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexptokenize ("aaabbb", "(.*ba).*") == [])
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexptokenize ("aaabbb", "(.*ba).*()") == [])
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** sformat **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
sformat ("%2 is greater %% than %1", 3, "five")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** conversions **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
tostring (tofloat (63), 4)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
tostring (0.12345, 4)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
tohexstring (31)
----------------------------------------------------------------------
