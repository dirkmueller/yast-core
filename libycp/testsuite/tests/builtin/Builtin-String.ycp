
# ---------------------------------------------------------
#
#  Filename:	Builtin-String.ycp
#
#  Purpose:	test cases for all string builtins
#
# ---------------------------------------------------------


("** size **")

(size (""))
(size ("linux"))


("** plus **")

("YaST" + "2")
("YaST" + 2)
("YaST" + .two)
("YaST" + `two)
{textdomain "base" ; return _("YaST") + "2" ; }
{textdomain "base" ; return "The " + _("YaST") ; }


("** equality **")

("YaST2" == "YaST2")
("YaST2" == "yast2")


("** inequality **")

("YaST2" != "yast2")
("YaST2" != "YaST2")


("** issubstring **")

(issubstring ("some text", "tex"))
(issubstring ("some text", "no"))


("** substring **")

(substring ("some text", 5))
(substring ("some text", 42))

(substring ("some text", 5, 2))
(substring ("some text", 42, 2))


("** mergestring **")

(mergestring (["", "abc", "dev", "ghi"], "/"))
(mergestring (["abc", "dev", "ghi", ""], "/"))
(mergestring ([1, "a", 3], "."))
