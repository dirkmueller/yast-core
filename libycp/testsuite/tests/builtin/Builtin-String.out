("** size **")
(0)
(5)
("** plus **")
("YaST2")
("YaST2")
("YaST.two")
("YaSTtwo")
("YaST2")
("The YaST")
("** equality **")
(true)
(false)
("** inequality **")
(true)
(false)
("** issubstring **")
(true)
(false)
("** substring **")
("text")
("")
("te")
("")
("** mergestring **")
("/abc/dev/ghi")
("abc/dev/ghi/")
(nil)
("")
("abcdevghi")
("abc123dev123ghi")
("** deletechars **")
("123")
("1a2b3c")
("")
("")
("** filterchars **")
("abc")
("")
("")
("")
("** crypt  -  disabled **")
("** tolower and toascii **")
("abcdef")
("aBc")
("** find et.al. **")
(4)
(-1)
(3)
(nil)
(2)
(nil)
(5)
(nil)
("** regex **")
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
("** sformat **")
("five is greater % than 3")
("** conversions **")
("63.0000")
("0.1235")
("0x1f")
